app/src/main/java/com/innowisegroup/datetimepicker/FragmentTimePicker.kt

```diff
@@ -43,53 +39,68 @@ class FragmentTimePicker : Fragment() {
         mHours?.wrapSelectorWheel = true
         mHours?.setFormatter { i: Int ->
             String.format(
-                    Locale.getDefault(),
-                    "%02d",
-                    i
+                Locale.getDefault(),
+                "%02d",
+                i
             )
         }
         mHours?.setDividerColor(
-                mHours,
-                ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
+            mHours,
+            ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
         )
-        mHours?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        mHours?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshTimeValue(
-                    localTime?.withHour(newVal)!!
+                localTime?.withHour(newVal)!!
             )
         }
 
         mMinutes?.minValue = MIN_MINUTE
         mMinutes?.maxValue = MAX_MINUTE
         mMinutes?.value = localTime?.getMinute()!!
         mMinutes?.setDividerColor(
-                mMinutes,
-                ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
+            mMinutes,
+            ResourcesCompat.getDrawable(resources, R.drawable.number_picker_divider_color, null)
         )
         mMinutes?.wrapSelectorWheel = true
         mMinutes?.setFormatter { i: Int ->
             String.format(
-                    Locale.getDefault(),
-                    "%02d",
-                    i
+                Locale.getDefault(),
+                "%02d",
+                i
             )
         }
-        mMinutes?.setOnValueChangedListener { _: NumberPicker?, _: Int, newVal: Int ->
+        mMinutes?.setOnValueChangedListener { _, _, newVal: Int ->
             refreshTimeValue(
-                    localTime?.withMinute(newVal)!!
+                localTime?.withMinute(newVal)!!
             )
         }
         return view
     }
 
     private fun refreshTimeValue(newValue: LocalTime) {
-        localTime = newValue
```

Comment:
return this line. now you have the incorrect value in the tab when you change the value in the picker